pyfar/dsp/dsp.py

```diff
@@ -811,6 +813,338 @@ def regularized_spectrum_inversion(
+        id_above = self._f_query > self._freq_range[1]
+
+        # interpolate the data
+        interpolated = []
+        for data in self._interpolators:
+            data_interpolated = np.concatenate((
+                (data[0](self._f_query[id_below])),
+                (data[1](self._f_query[id_within])),
+                (data[2](self._f_query[id_above]))),
+                axis=-1)
+            interpolated.append(data_interpolated)
+
+        # get half sided spectrum
+        if self._method == "complex":
+            freq = interpolated[0] + 1j * interpolated[1]
+        elif self._method == 'magnitude_unwrap':
+            freq = interpolated[0] * np.exp(-1j * interpolated[1])
+        else:
+            freq = interpolated[0]
+
+        # get initial signal
+        signal = pyfar.Signal(freq, sampling_rate, n_samples, "freq")
+
+        # clip the magnitude
+        if self._clip:
+            signal.freq = \
+                np.clip(np.abs(signal.freq), self._clip[0], self._clip[1]) \
+                * np.exp(-1j * phase(signal))
+
+        # generate linear or minimum phase
+        if self._method == "magnitude_minimum":
+            pass
+        elif self._method == "magnitude_linear":
+            signal = linear_phase(signal, self._group_delay, self._unit)
+
+        if show:
+            # plot input and output data
+            with pyfar.plot.context():
+                _, ax = plt.subplots(2, 2)
+                # time signal (linear amplitude)
+                pyfar.plot.time(signal, ax=ax[0][0])
+                # time signal (log amplitude)
+                pyfar.plot.time(signal, ax=ax[1][0], dB=True)
+                # frequency plot (linear x-axis)
+                pyfar.plot.freq(signal, dB=False, xscale="linear", ax=ax[0][1])
+                pyfar.plot.freq(self._input, dB=False, xscale="linear",
+                                ax=ax[0][1], c='r', ls='', marker='.')
+                ax[0][1].set_xlim(0, sampling_rate/2)
+                # frequency plot (log x-axis)
+                pyfar.plot.freq(signal, dB=False, ax=ax[1][1], label='intput')
+                pyfar.plot.freq(self._input, dB=False, ax=ax[1][1],
+                                c='r', ls='', marker='.', label='output')
+                min_freq = np.min([sampling_rate / n_samples,
+                                   self._input.frequencies[0]])
+                ax[1][1].set_xlim(min_freq, sampling_rate/2)
+                ax[1][1].legend(loc='best')
+
+        return signal
+
+    def _get_frequencies(self, frequencies):
+        """
+        Return frequencies for creating or quering interpolation objects.
+
+        In case logfrequencies are requested, 0 Hz entries are replaced by
+        the next highest frequency, because the logarithm of 0 does not exist.
+        """
+        if self._fscale == "log":
+            if frequencies[0] == 0:
+                frequencies[0] = frequencies[1]
+            frequencies = np.log(frequencies)
```

Comment:
I'd change the behavior of this to set the magnitude at 0 Hz to 0, since this would be a static sound pressure, or dc voltage.
But this is personal preference, since I almost always deal with signals which are DC free

Reply:
I thought about this too. There are examples where the current version works better, e.g., interpolating reflexion coefficients of walls, and where your suggestion works better, e.g., interpolating HRTFs of room transfer functions. I thought about making this an optional input parameter but did not due it to keep things simlpe and because changing the value afterwards is easy (`signal.freq[:::, 0]=0`).

Reply:
Makes sense